### PR TITLE
Move analysis components into a package

### PR DESCRIPTION
**Pull request by maxrjones**

### CODE REVIEW

File: main.sh [hunk 1]
```diff
@@ -1,4 +1,11 @@
 #!/bin/bash
 set -e
 playwright install
-python main.py
+python main.py --dataset 1MB-chunks --zarr-version v2 --action zoom_in --zoom-level 4 --s3-bucket s3://carbonplan-benchmarks
```
Comment (andersy005) [inline main.sh]: should we enable the `--detect-provider` flag so we can track where the results are recorded from?

Comment (maxrjones) [reply main.sh]: good call, thanks!

File: main.sh [hunk 2]
```diff
@@ -3,4 +3,11 @@ set -e
...
 playwright install
-pytest -v -s
+python main.py --dataset 1MB-chunks --zarr-version v2 --action zoom_in --zoom-level 4 --s3-bucket s3://carbonplan-benchmarks --detect-provider
+python main.py --dataset 5MB-chunks --zarr-version v2 --action zoom_in --zoom-level 4 --s3-bucket s3://carbonplan-benchmarks --detect-provider
+python main.py --dataset 10MB-chunks --zarr-version v2 --action zoom_in --zoom-level 4 --s3-bucket s3://carbonplan-benchmarks --detect-provider
+python main.py --dataset 25MB-chunks --zarr-version v2 --action zoom_in --zoom-level 4 --s3-bucket s3://carbonplan-benchmarks --detect-provider
+python main.py --dataset 1MB-chunks --zarr-version v3 --action zoom_in --zoom-level 4 --s3-bucket s3://carbonplan-benchmarks --detect-provider
+python main.py --dataset 5MB-chunks --zarr-version v3 --action zoom_in --zoom-level 4 --s3-bucket s3://carbonplan-benchmarks --detect-provider
+python main.py --dataset 10MB-chunks --zarr-version v3 --action zoom_in --zoom-level 4 --s3-bucket s3://carbonplan-benchmarks --detect-provider
+python main.py --dataset 25MB-chunks --zarr-version v3 --action zoom_in --zoom-level 4 --s3-bucket s3://carbonplan-benchmarks --detect-provider
```
Comment (andersy005) [inline main.sh]: turn this into loops??

```bash
datasets=("1MB-chunks" "5MB-chunks" "10MB-chunks" "25MB-chunks")
versions=("v2" "v3")

for version in "${versions[@]}"
do
  for dataset in "${datasets[@]}"
  do
    python main.py --dataset $dataset --zarr-version $version --action zoom_in --zoom-level 4 --s3-bucket s3://carbonplan-benchmarks --detect-provider
  done
done
```

Comment (maxrjones) [reply main.sh]: nice! thanks

**Comment by maxrjones** [comment]
@andersy005 are you alright with making the analysis components into a package? I don't think a full review is needed, but am interested in your take on that change before merging

**Comment by andersy005** [comment]
> @andersy005 are you alright with making the analysis components into a package? I don't think a full review is needed, but am interested in your take on that change before merging

👍🏽this sounds great to me, max. i also support moving main.py inside the package as well. happy to provide feedback/review and/or move main inside the package.

**Pull request merged**